CMakeLists.txt

```diff
@@ -661,12 +661,16 @@ if(BUILD_TESTS)
   endif()
 
   if(TLS_TEST)
-    add_custom_target(
-      testssl ALL
+    add_custom_command(
+      OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/testssl/testssl.sh
       COMMAND
-        test -d testssl || git clone https://github.com/drwetter/testssl.sh
+        rm -rf ${CMAKE_CURRENT_BINARY_DIR}/testssl && git clone --depth 1
+        https://github.com/drwetter/testssl.sh
         ${CMAKE_CURRENT_BINARY_DIR}/testssl
     )
+    add_custom_target(
+      testssl ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/testssl/testssl.sh
+    )
```

Comment:
The beahaviour is the same (there is a `testssl` target which clones the repo to get a local `testssl/testssl.sh` file). But with the additional custom target indirection, we only build that when we need to. Previously, whenever `TLS_TEST` was on, we'd build the `testssl` target every time, and rebuild additional targets whenever it was turned on or off.